extensions/data-plane/data-plane-http/src/main/java/org/eclipse/edc/connector/dataplane/http/pipeline/HttpRequestParamsSupplier.java

```diff
@@ -30,9 +32,11 @@
 public abstract class HttpRequestParamsSupplier implements Function<DataFlowRequest, HttpRequestParams> {
 
     private final Vault vault;
+    private final TypeManager typeManager;
 
-    protected HttpRequestParamsSupplier(Vault vault) {
+    protected HttpRequestParamsSupplier(Vault vault, TypeManager typeManager) {
         this.vault = vault;
+        this.typeManager = typeManager;
```

Comment:
we're not using any functionality specifically from the `TypeManager`, so using _just_ an `ObjectMapper` would be better IMO.

Reply:
Although we heavily rely on jackson (it actually can't be replaced), having an dedicated wrapper class is useful as it provides a neater interface and makes our services talk about our language, not the library's one.

Reply:
The `TypeManager` is more than just a wrapper for an `ObjectMapper`, it handles type registration, delegation to multiple serialization contexts, etc.

I would thus argue, that its interface contains methods that have nothing to do with just serialization and deserialization, the Oauth2 module does not need multiple serialization contexts, or care about type registrations.

As you said, we cannot easily replace Jackson anyway, so why not just embrace it and use the `ObjectMapper` directly. It would be a more focused class.

Reply:
Since this approach is widely used I think we should take a clear decision and say, for example "never pass TypeManager to services, use ObjectMapper instead" and then change this everywhere.

Reply:
> Since this approach is widely used I think we should take a clear decision and say, for example "never pass TypeManager to services, use ObjectMapper instead" and then change this everywhere.

Absolutely agree, and IIRC there was already talk about this in the past, I think in the context of the SQL stores. Don't recall whether it was a comment on a PR, or a discussion, but that somehow rings a bell. 
How about we come up with some sort of "coding guidelines", which would clearly state that, amongst many others?

Reply:
I don't know, I always liked the `TypeManager` approach :grin: but it could be.
I'm ok for addressing this to another issue.

Reply:
Frankly, having coding guidelines is something that's been lurking in the back of my mind for quite a while now, and I have already started to write some, but abandoned it due to a lack of time. Maybe I can dig out that branch once again.